x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -95,6 +110,278 @@
+                            .field(ENABLED, false)
+                        .endObject()
+                        .startObject(CategorizationAnalyzerConfig.CHAR_FILTERS.getPreferredName())
+                            .field(ENABLED, false)
+                        .endObject()
+                    .endObject()
+                .endObject()
+                .startObject(AnalysisConfig.LATENCY.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(AnalysisConfig.SUMMARY_COUNT_FIELD_NAME.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(AnalysisConfig.DETECTORS.getPreferredName())
+                    .startObject(PROPERTIES)
+                        .startObject(Detector.DETECTOR_DESCRIPTION_FIELD.getPreferredName())
+                            .field(TYPE, TEXT)
+                        .endObject()
+                        .startObject(Detector.FUNCTION_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.BY_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.OVER_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.PARTITION_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.USE_NULL_FIELD.getPreferredName())
+                            .field(TYPE, BOOLEAN)
+                        .endObject()
+                        .startObject(Detector.EXCLUDE_FREQUENT_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.CUSTOM_RULES_FIELD.getPreferredName())
+                            .field(TYPE, NESTED)
+                            .startObject(PROPERTIES)
+                                .startObject(DetectionRule.ACTIONS_FIELD.getPreferredName())
+                                    .field(TYPE, KEYWORD)
+                                .endObject()
+                                // RuleScope is a map
+                                .startObject(DetectionRule.SCOPE_FIELD.getPreferredName())
+                                    .field(ENABLED, false)
+                                .endObject()
+                                .startObject(DetectionRule.CONDITIONS_FIELD.getPreferredName())
+                                    .field(TYPE, NESTED)
```

Comment:
I think at this level `NESTED` is definitely overkill and should be `OBJECT`.

Reply:
Again conditions are an array. It's arguable that the extensive mappings are overkill anyway, not all fields need to be searchable and these mappings come with a maintenance burden. Is it conceivable that someone will want to search for a job where `RuleCondition.AppliesTo == ACTUAL && RuleCondition.Operator == GT`

Reply:
I guess there will be relatively few documents in the index, so we can get away with high-cost mappings.  We have `NESTED` in our results index where it is much more costly due to the higher document volume and it's never been found to be the root cause of any problem, so it's probably fine.